src/components/document/sort-work-view.tsx

```diff
@@ -75,6 +75,10 @@ export const SortWorkView: React.FC = observer(function SortWorkView() {
   const openDocumentKey = tabState?.openDocuments.get(ENavTab.kSortWork) || "";
   const showSortWorkDocumentArea = !!openDocumentKey;
 
+  const openDocumentsGroup = openDocumentKey
+    ? sortedDocumentGroups.find(group => group.documents.some(doc => doc.key === openDocumentKey)) ?? undefined
+    : undefined;
+
```

Comment:
I think this will not work in all cases. Some documents can be repeated across multiple groups. For example when sorting by tools. In that case the `openDocumentsGroup` will be the first group the document is part of instead of the group that the user actual opened document in. Because of this I think when a document is opened, the code needs to add information about that group to the persistent UI.

The persistent UI has a second level of tabs which could be used used for this. That second level is why you have to do: `tabState?.openDocuments.get(ENavTab.kSortWork)`. So this second level of tabs could be used to indicate the sort group that the document is inside of. That would mean instead of hardcoding the `openDocuments.get(ENavTab.kSortWork)` it would be `openDocuments.get(tabState.openSubTab)`. And when the document is opened, `persistentUI.openSubTabDocument(...)` should be called with a `subTab` parameter of the sortedDocumentGroup that it is part of.

Here is my proposal for the identifier used as the `openSubTab`. Encode the primary and second sort into a single identifier. So if it is just sorting by name then the string would be `"name"`. If it was sorting by name and then by tools the string would be `"name.tools"`.

There is one more feature of the persistentUI that we'll have to deal with when taking this approach. It is designed to track an open document in each of the sub tabs. This way if you navigate between sub tabs you'll see the same document when you go back to a previous sub tab. In the case of the sort work tab, this doesn't match. The sort work tab only has one open document. What this means is that after calling `openSubTabDocument` we should clear out the other documents in `tabState.openDocuments`. There is a `closeSubTabDocument` action that can be used for this.

I'd guess that I'll need to provide more info about all this, so don't hesitate to ask questions. We can also meet about it tomorrow.